WordPress/Classes/ViewRelated/Reader/ReaderDetailViewController.swift

```diff
@@ -590,11 +590,16 @@ open class ReaderDetailViewController: UIViewController, UIViewControllerRestora
         }
 
         // Byline
-        let date = NSDate(timeIntervalSinceReferenceDate: (post?.dateForDisplay().timeIntervalSinceReferenceDate)!)
-        var byline = date.mediumString()
-        if let author = post?.authorForDisplay() {
-            byline = String(format: "%@ · %@", author, byline)
+        let author = post?.authorForDisplay()
+        let dateAsString = post?.dateForDisplay()?.mediumString()
+        let byline: String
+
+        if let author = author, let date = dateAsString {
+            byline = author + " · " + date
+        } else {
+            byline = author ?? dateAsString ?? String()
         }
+
```

Comment:
If I were to nitpick, I'd ask if there was a way to do this that avoided the possibility of assigning nils to consts. (Call it a pet peeve.)  Something like this maybe?  
```
var byline = ""
        if let author = post?.authorForDisplay() {
            byline = author
        }
        if let dateAsString = post?.dateForDisplay()?.mediumString() {
            if byline.characters.count > 0 {
                byline += " . "
            }
            byline += dateAsString
        }
```

But what you have looks solid.